modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/impl/CommerceCurrencyLocalServiceImpl.java

```diff
@@ -405,6 +409,9 @@ public CommerceCurrency updateCommerceCurrency(
 			rate = BigDecimal.ONE;
 		}
 
+		_validateExternalReferenceCode(
+			externalReferenceCode, commerceCurrency.getCommerceCurrencyId(),
+			serviceContext.getCompanyId());
```

Comment:
there is no need to implement a validator for duplication control because it is automatically generated see [here](https://github.com/victorcdp/liferay-portal/blob/7f63ecc3e6b2c30122aea51cb9cbf2649c11ea62/modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/service/persistence/impl/CommerceCurrencyPersistenceImpl.java#L4098)